From 206aa475636ae8a19fa4d3bb1d938e8cfe9a7dd2 Mon Sep 17 00:00:00 2001
From: Park Ju Hyung <qkrwngud825@gmail.com>
Date: Wed, 22 Mar 2017 11:41:21 +0900
Subject: [PATCH 5/6] aapt: enforce 0 compression

In extension to the commit f0e2610c81d(aapt: default to 0 compression),
this commit now enforces 0 compression for every contents inside an apk.

Change-Id: Idd5a1b4da9f35f7a29f5595ed51fcb2e66b73816
Signed-off-by: Park Ju Hyung <qkrwngud825@gmail.com>
---
 tools/aapt/Android.mk        | 2 +-
 tools/aapt/Main.cpp          | 2 +-
 tools/aapt/Package.cpp       | 7 ++++++-
 tools/aapt/Resource.cpp      | 4 ++++
 tools/aapt/ResourceTable.cpp | 6 +++++-
 5 files changed, 17 insertions(+), 4 deletions(-)

diff --git a/tools/aapt/Android.mk b/tools/aapt/Android.mk
index 250fffd8eb3..a94f5effac9 100644
--- a/tools/aapt/Android.mk
+++ b/tools/aapt/Android.mk
@@ -68,7 +68,7 @@ aaptCFlags := -DAAPT_VERSION=\"$(BUILD_NUMBER_FROM_FILE)\"
 aaptCFlags += -Wall -Werror
 
 ifeq ($(TARGET_WANTS_AAPT_COMPRESS),true)
-aaptCFlags += -DAAPT_COMPRESS=1
+aaptCFlags += -DAAPT_COMPRESS
 endif
 
 aaptHostLdLibs_linux := -lrt -ldl -lpthread
diff --git a/tools/aapt/Main.cpp b/tools/aapt/Main.cpp
index 6019f76afdf..8087660e9d5 100644
--- a/tools/aapt/Main.cpp
+++ b/tools/aapt/Main.cpp
@@ -269,7 +269,7 @@ int main(int argc, char* const argv[])
     int tolerance = 0;
 
     /* default to 0 compression, unless the target opted out*/
-#if AAPT_COMPRESS
+#ifdef AAPT_COMPRESS
     bundle.setCompressionMethod(ZipEntry::kCompressDeflated);
 #else
     bundle.setCompressionMethod(ZipEntry::kCompressStored);
diff --git a/tools/aapt/Package.cpp b/tools/aapt/Package.cpp
index d631f353112..a67fbba2b4d 100644
--- a/tools/aapt/Package.cpp
+++ b/tools/aapt/Package.cpp
@@ -323,11 +323,16 @@ bool processFile(Bundle* bundle, ZipFile* zip,
     if (fromGzip) {
         result = zip->addGzip(file->getSourceFile().string(), storageName.string(), &entry);
     } else if (!hasData) {
+        int compressionMethod;
+#ifdef AAPT_COMPRESS
         /* don't compress certain files, e.g. PNGs */
-        int compressionMethod = bundle->getCompressionMethod();
+        compressionMethod = bundle->getCompressionMethod();
         if (!okayToCompress(bundle, storageName)) {
             compressionMethod = ZipEntry::kCompressStored;
         }
+#else
+        compressionMethod = ZipEntry::kCompressStored;
+#endif
         result = zip->add(file->getSourceFile().string(), storageName.string(), compressionMethod,
                             &entry);
     } else {
diff --git a/tools/aapt/Resource.cpp b/tools/aapt/Resource.cpp
index 1437766e699..76e28cd79a2 100644
--- a/tools/aapt/Resource.cpp
+++ b/tools/aapt/Resource.cpp
@@ -1219,7 +1219,11 @@ status_t generateAndroidManifestForSplit(Bundle* bundle, const sp<AaptAssets>& a
     if (err < NO_ERROR) {
         return err;
     }
+#ifdef AAPT_COMPRESS
     outFile->setCompressionMethod(ZipEntry::kCompressDeflated);
+#else
+    outFile->setCompressionMethod(ZipEntry::kCompressStored);
+#endif
     return NO_ERROR;
 }
 
diff --git a/tools/aapt/ResourceTable.cpp b/tools/aapt/ResourceTable.cpp
index 4200fd51305..24582d87f92 100644
--- a/tools/aapt/ResourceTable.cpp
+++ b/tools/aapt/ResourceTable.cpp
@@ -144,8 +144,12 @@ status_t compileXmlFile(const Bundle* bundle,
         printXMLBlock(&tree);
     }
 
+#ifdef AAPT_COMPRESS
     target->setCompressionMethod(ZipEntry::kCompressDeflated);
-    
+#else
+    target->setCompressionMethod(ZipEntry::kCompressStored);
+#endif
+
     return err;
 }
 
-- 
2.11.0

